### DISCUSSION TITLE
Run pocketbase directly from mobile

### DISCUSSION THREAD

**Discussion by rohitsangwan01** [pocketbase/pocketbase#2611] [Show and tell]
Now pocketbase is not only limited to desktop, we can run this awesome project directly from our mobiles as well by using my application which we can install directly from play store , 

https://play.google.com/store/apps/details?id=com.pocketbase.mobile

Here are complete details of the project
https://rohitsangwan.hashnode.dev/run-pocketbase-from-mobile

i have no idea where to share or discuss with community for feedback or improvements, so help would be really appreciated !

<img src="https://github.com/rohitsangwan01/pocketbase_server_flutter/assets/59526499/7d20a2a4-0df7-4f2a-90bf-2577289e0f7e" height="300">
<img src="https://github.com/rohitsangwan01/pocketbase_server_flutter/assets/59526499/370c007d-51c3-45a9-928c-1287c8def0d3" height="300">
<img src="https://github.com/rohitsangwan01/pocketbase_server_flutter/assets/59526499/657a6e4c-8431-4f49-b29d-a0f599524f6c" height="300">
<img src="https://github.com/rohitsangwan01/pocketbase_server_flutter/assets/59526499/4ecd5f1c-ae2b-4406-a10d-0d9ae3e9900e" height="300">
<img src="https://github.com/rohitsangwan01/pocketbase_server_flutter/assets/59526499/f58f7f5e-d3d0-4328-a8be-f5cf12e15cdb" height="300">

**Comment by gregoryforel** [comment]
That's great! So it would be possible to embed or wrap pocketbase in a flutter app and use it instead of sqflite or drift for example?

**Comment by rohitsangwan01** [comment]
@gregoryforel well sqflite or drift are for using a database locally, pocketbase will not fit for this case in flutter, because this will be used like a backend, and we have to connect with that using the local url , which i think is not good option compared to directly using a local database

**Comment by gregoryforel** [comment]
I see, but I thought there would have been a way to actually run the
backend on the phone and query it directly locally. Like the way NodeJs is
self contained in an electron app.

**Reply by rohitsangwan01** [reply]
Yes that we can do with Pocketbase too, here is my example flutter app
https://github.com/rohitsangwan01/pocketbase_server_flutter

but i can release a flutter library for using Pocketbase server directly in flutter mobile, if  really required by community

**Reply by gregoryforel** [reply]
Yes, this is very good. I think PocketBase is on its way to making A LOT of
noise in the BAAS ecosystem. I can foresee Flutter offline-first Desktop /
Mobile apps developed with an embedded PB server, synchronized to a remote
PB server.

**Comment by rohitsangwan01** [comment]
@gregoryforel Just released it on Pub.dev to directly use in Flutter Projects 

https://pub.dev/packages/pocketbase_server_flutter

**Reply by gregoryforel** [reply]
> @gregoryforel Just released it on Pub.dev to directly use in Flutter Projects 
> 
> https://pub.dev/packages/pocketbase_server_flutter

I'll test that soon, thank you!

**Reply by tigawanna** [reply]
Can this be implemented in react native?

**Reply by rohitsangwan01** [reply]
> Can this be implemented in react native?

This is for flutter, but you can use the native libraries for react native implementation 

[pocketbase_server_flutter](https://github.com/rohitsangwan01/pocketbase_server_flutter) to use in Flutter

[pocketbase_android](https://github.com/rohitsangwan01/pocketbase_android) to use in native Android
 
[pocketbase_ios](https://github.com/rohitsangwan01/pocketbase_ios) to use in native IOS

**Reply by tigawanna** [reply]
Awesome

**Comment by Dokotela** [comment]
@rohitsangwan01 this looks awesome. And I wanted to ask about a use case I have. I'm a physician and Flutter developer. One of the projects we're interested in is an application for emergency relief work internationally. On the ground, we couldn't rely on any kind of network or internet except what we could provide (there are usually generators, so there is electricity). What I would love is a single mobile device that would serve as a LAN and ran PocketBase. Other devices would then have a user application that would communicate with the main device. With what you've created in the pocketbase_server_flutter, do you think this would be possible?

**Reply by SamuelMarks** [reply]
@Dokotela Definitely possible. About to expand into the same area; the TL;DR is to use CRDTs to synchronise and that way you can have part/most of the network offline indefinitely. And still have each node be completely functional. Happy to chat more about how to build this; and will be open-sourcing our solutions so are probably open to collaboration (Harvard Medical - simarks[at]meei.harvard.edu)

**Reply by rohitsangwan01** [reply]
@Dokotela yes i think pocketbase_server_flutter is good fit for this usecase

**Reply by Dokotela** [reply]
@rohitsangwan01 I've gotten around to trying this out. I have a version of pocketbase that I'd like to use, but it's not the base image. Is there a way to use my own pocketbase executable and fun it in flutter?

**Reply by rohitsangwan01** [reply]
@Dokotela not yet, you have to build on your own
